src/psyclone/tests/psyir/transformations/arrayassignment2loops_trans_test.py

```diff
@@ -499,18 +499,23 @@ def test_character_validation(fortran_reader):
         " by default (use the'allow_string' option to expand them), but found"
         in str(info.value))
 
-    # TODO #2612: We can not identify cases with 'character(LEN=x)' syntax as
-    # this is currently an UnsupportedFortranType
+
+def test_unsupported_type_character(fortran_reader):
     code = '''subroutine test()
```

Comment:
Please add a docstring. Should we keep a reference to #2612, perhaps have an xfailing bit at the end of this test for e.g. `character(LEN=max_len) :: c`

Reply:
Added docstring. The declaration you mention still succeeds, so I mentioned the issue in the docstring for reference but I don't make it a full TODO, as this test shouldn't change if we ever convert it from a partial type to a full type.